### DISCUSSION TITLE
V2.2.8 shows strange graphs

### DISCUSSION THREAD

**Discussion by ClausSamuelsen** [UtilitechAS/amsreader-firmware#449] [General]
After upgrading to v2.2.8 the graphs for last 24 hours and last month is completely wrong.
Never seen this in previous versions.
Is this a bug i v2.2.8 or must I reconfigure something.
Data sent to MQTT is fine.
<img width="1525" alt="amsreader-v2 2 8" src="https://user-images.githubusercontent.com/96246629/218972626-12b5efbc-ca02-4dc4-a47e-b3281c221f13.png">
Best regards
Claus

**Comment by ArnieO** [comment]
Most of us upgraded without any such issue, and we have more than 1000 downloads of v2.2.x now. 
I think you can fix this by exporting a config file, editing it then reading it back to the device.

**Comment by gskjold** [comment]
Did you upgrade from a previous 2.2.x or from 2.1?

**Reply by ClausSamuelsen** [reply]
I upgraded from v2.2.7

**Reply by gskjold** [reply]
Interesting. There are only GUI changes from 2.2.7 to 2.2.8, so this must be some edge case I have yet to identify

**Comment by ClausSamuelsen** [comment]
I had an old config file v2.1.14, and comparing with v2.2.8 the most obvious differences are in the dayplot and monthplot values

amsconfig
version v2.1.14
...
thresholds 5 10 15 20 25 50 75 100 150 255 3
dayplot 4 1670079600 40434332 1050 1050 1050 1050 1050 1050 1050 1050 1050 1050 1050 1050 1050 1050 1050 1160 1380 1210 1050 1050 1050 1050 1050 1050
monthplot 5 1669935604 40391930 26060 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 8840 10230 12930 12770 11920 12340 18310 20460 16730 13130 13250 18060 9720 28370 0
energyaccounting 4 12 0.00 0.00 0.00 0.00 1 1.72 2 1.47 0 0.00 0 0.00 0 0.00

amsconfig
version v2.2.8
...
thresholds 5 10 15 20 25 50 75 100 150 255 3
dayplot 5 1676458803 42219110 3 0 0 0 0 0 0 0 37994000 1000 0 1000 0 0 0 0 0 0 0 0 0 0 0 0 0
monthplot 6 1676415603 4221027 5 0 0 0 0 0 0 0 0 0 0 0 0 -37967296 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
energyaccounting 5 2 0.90 12 0 0.00 0 0 7 1.87 2 2.07 15 638.48 11 2.20 5 1.73

**Reply by gskjold** [reply]
Dayplot, monthplot and energyaccounting all have structural changes between 2.1 and 2.2. But obviously the numbers also differs, as I guess it is some time ago that you hade 2.1.14

**Comment by ClausSamuelsen** [comment]
Is there a method for resetting the meter data, other than a factory reset?

**Reply by ArnieO** [reply]
You can edit the *.cnf file and upload it back to the device.

**Reply by ClausSamuelsen** [reply]
I edited the config file like this:
dayplot 5 1676476803 42225490 3 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
monthplot 6 1676415603 4221027 5 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0

So there's 24 zeroes in dayplot and 31 zeroes in monthplot.
Should I also change the first 4 values in each line (5 1676476803 42225490 3 and 6 1676415603 4221027 5) ?
Can you describe what the first 4 values are?

**Reply by ClausSamuelsen** [reply]
And what about the energyaccounting - should that also be zeroed out?

**Reply by gskjold** [reply]
first value is a version, then timestamp for last meter update and then the kwh for last update. I would leave those and check that the graphs picks up where it left off. I see that energyaccounting is also wrong on your device. I would go for:
energyaccounting 5 2 0.0 0 0 0.00 0 0 0 0.00 0 0.00 0 0.0 0 0.0 0 0.00
which would reset all of it

**Reply by ClausSamuelsen** [reply]
After zeroing out dayplot and monthplot, but not energyaccounting, the hourly consumption now looks fine (after 15 hours), but monthplot values (actually only one value) is wrong.
I have now also zeroed energyaccounting, so will come back within a couple of days, when all figures hopefully looks correct.
Thanks for helping.
Regards
Claus

**Reply by ClausSamuelsen** [reply]
Just want to say, that all values in the graphs are ok.
Thanks for helping
Claus
